lib/dispatch.go

```diff
@@ -411,3 +416,50 @@ func dispatchReturnError(got interface{}, err error) error {
 	}
 	return err
 }
+
+// normalizeInputParams will look at each field of the params, and modify filepaths so that
+// they are absolute paths, making them safe to send across RPC to another process
+func normalizeInputParams(param interface{}) interface{} {
+	typ := reflect.TypeOf(param)
+	val := reflect.ValueOf(param)
+	if typ.Kind() == reflect.Ptr {
+		typ = typ.Elem()
+		val = val.Elem()
+	}
+	if typ.Kind() == reflect.Struct {
+		num := typ.NumField()
+		for i := 0; i < num; i++ {
+			tfield := typ.Field(i)
+			vfield := val.Field(i)
+			qriTag := tfield.Tag.Get("qri")
+			if qriTag == "fspath" || qriTag == "dsrefOrFspath" {
```

Comment:
flip these into constants & add a comment describing the `qri` struct tag:

```
qri struct tags augment the way the qri runtime interprets a field
```

then add a comment on each constant describing how it will augment normalization. Your PR description lays both of these out nicely 😄

Reply:
👍